DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -200,6 +200,32 @@ private async ValueTask<TokenKind> GetTokenKindAsync(CancellationToken cancellat
         return Guild.Parse(response);
     }
 
+    public async IAsyncEnumerable<ThreadChannel> GetGuildChannelThreadsAsync(
+        string channelId,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        int currentOffset = 0;
+
+        while (true)
+        {
+            var url = new UrlBuilder()
+                .SetPath($"channels/{channelId}/threads/search")
+                .SetQueryParameter("offset", currentOffset.ToString())
+                .Build();
+
+            var response = await TryGetJsonResponseAsync(url, cancellationToken);
+
+            foreach(var threadJson in response?.GetProperty("threads").EnumerateArray() ?? Enumerable.Empty<JsonElement>())
+                yield return ThreadChannel.Parse(threadJson);
+
+            if (!(response?.GetProperty("has_more").GetBoolean() ?? false))
+            {
+                break;
+            }
+                currentOffset += 25;
```

Comment:
We should probably not rely on a fixed number here and instead increment but the number of items that were returned in the search

Reply:
Resolved in 6183c87.

As with the thread parsing section, I got a few errors before I inserted the null-coalescing operator at the end:
```
CS0266 Cannot implicitly convert type 'int?' to 'int'. An explicit conversion exists (are you missing a cast?)
CS8629 Nullable value type may be null.
```
Still not sure if this sort of thing is the best thing to be doing, but it made the red line go away and the program seems to work fine.